crates/ruma-events/src/member_hints.rs

```diff
@@ -0,0 +1,98 @@
+//! Types for Matrix member hint state events ([MSC4171]).
+//!
+//! This implements `m.member_hints` state event described in [MSC4171].
+//!
+//! [MSC4171]: https://github.com/matrix-org/matrix-spec-proposals/pull/4171
+
+use std::collections::BTreeSet;
+
+use ruma_common::OwnedUserId;
+use ruma_events::content::EventContent;
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    EmptyStateKey, PossiblyRedactedStateEventContent, RedactContent, RedactedStateEventContent,
+    StateEventType,
+};
+
+/// The content for an `m.member_hints` state event.
+///
+/// Any users (service members) listed in the content should not be considered when computing the
+/// room name or avatar based on the member list.
+#[derive(Clone, Debug, Default, Serialize, Deserialize, EventContent, PartialEq)]
+#[ruma_event(type = "io.element.functional_members", kind = State, state_key_type = EmptyStateKey,
+custom_redacted, custom_possibly_redacted)]
```

Comment:
It doesn't seem like you need to use `custom_redacted`.

Reply:
Ah indeed, thanks. https://github.com/ruma/ruma/pull/1952/commits/d7ad8a3a0e6d38391ef6ef4a298b759da7733d1b